src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,264 @@
+"""
+This script creates a single table of sample QC & batching metrics from EvidenceQC.
+"""
+
+import argparse
+import pandas as pd
+from collections import Counter
+from numpy import array
+from functools import reduce
+
+
+ID_COL = "#ID"
+
+
+def read_ploidy(filename: str) -> pd.DataFrame:
+    """
+
+    Args:
+        filename: A tab-delimited file containing estimated copy numbers.
+
+    Returns:
+        A pandas DataFrame containing the following columns:
+        [id, chr1_CopyNumber, ..., chr22_CopyNumber, chrX_CopyNumber, chrY_CopyNumber, chrX_CopyNumber_rounded].
+    """
+    df_ploidy = pd.read_csv(filename, sep="\t")
+    df_ploidy.loc[round(df_ploidy["chrX_CopyNumber"]) < 2, "chrX_CopyNumber_rounded"] = 1
+    df_ploidy.loc[round(df_ploidy["chrX_CopyNumber"]) >= 2, "chrX_CopyNumber_rounded"] = 2
+    return df_ploidy
+
+
+def read_bincov_median(filename: str) -> pd.DataFrame:
+    """
+    Median coverage (bincov_median)
+    Args:
+        filename: bincov_median output from EvidenceQC (medianCov.transposed) in a tab separated format
+
+    Returns:
+        a pandas DataFrame containing the transposed table given in the input file.
+    """
+    df_median = pd.read_csv(filename, sep="\t").T
+    df_median = df_median.rename_axis(ID_COL).reset_index()
+    df_median.columns = [ID_COL, "median_coverage"]
+    df_median = df_median.reset_index(drop=True)
+    return df_median
+
+
+def read_wgd_scores(filename: str) -> pd.DataFrame:
+    return pd.read_csv(filename, sep="\t")
+
+
+def read_non_diploid(filename: str) -> pd.DataFrame:
+    """
+    EvidenceQC → untar → /ploidy_est/binwise_CNV_qValues.bed.gz →
+    count number of bins with q-value < 0.05 for each sample
+
+    Args:
+        filename: a tab-delimited file containing binwise CNV qValues.
+
+    Returns:
+        A pandas DataFrame containing the number of bins with q-value < 0.05 for each sample.
+    """
+    df_non_diploid = pd.read_csv(filename, sep="\t").T
+    df_non_diploid = df_non_diploid.T.iloc[0:, 3:]
+    new_c = df_non_diploid.iloc[0:, 0:]
+    nondiploid_counts = new_c[new_c < 0.05].count()
+    nondiploid_counts_df = pd.DataFrame(nondiploid_counts, columns=["nondiploid_counts"]).rename_axis(
```

Comment:
Let's name this column `nondiploid_bins` to indicate where it came from
```suggestion
    nondiploid_counts_df = pd.DataFrame(nondiploid_counts, columns=["nondiploid_bins"]).rename_axis(
```